src/main.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+extern crate env_logger;
+#[macro_use] extern crate failure;
+extern crate getopts;
+extern crate sandboxfs;
+
+use failure::Error;
+use getopts::Options;
+use std::env;
+use std::path::Path;
+use std::process;
+use std::result::Result;
+
+/// Execution failure due to a user-triggered error.
+#[derive(Debug, Fail)]
+#[fail(display = "{}", message)]
+struct UsageError {
+    message: String,
+}
+
+/// Obtains the program name from the execution's first argument, or returns a default if the
+/// program name cannot be determined for whatever reason.
+fn program_name(args: &[String], default: &'static str) -> String {
+    let default = String::from(default);
+    match args.get(0) {
+        Some(arg0) => match Path::new(arg0).file_name() {
+            Some(basename) => match basename.to_str() {
+                Some(basename) => String::from(basename),
+                None => default,
+            },
+            None => default,
+        },
+        None => default,
+    }
+}
```

Comment:
Personally I find using the methods on Option much easier to understand: https://rust.godbolt.org/z/sGTXht

You can also return a `&str` to save an allocation but it probably isn't worth it in this case: https://rust.godbolt.org/z/7wGCfT